snap/snapcraft.yaml

```diff
@@ -49,52 +47,25 @@ apps:
     plugs: [network, network-bind]
 
 parts:
-  # This go part is necessary because it's expected that this snapcraft.yaml
-  # will be build on Linux Foundation infrastructure and as such runs in a 
-  # docker container. Since this may be run in a docker container, we can't 
-  # use the default (and convenient) `build-snaps: [go/1.15]` inside the
-  # device-simple part which requires a working snapd which isn't available
-  # inside a docker container
-  go:
+  version:
     plugin: nil
     source: snap/local
-    build-packages: [curl]
-    override-build: |
-      # Use `dpkg architecture` to figure out our target arch rather than 
-      # using `arch` because this allows cross compiling from arm64 to target
-      # armhf, etc.
-      case "$(dpkg --print-architecture)" in
-        amd64)
-          FILE_NAME=go1.16.4.linux-amd64.tar.gz
-          FILE_HASH=7154e88f5a8047aad4b80ebace58a059e36e7e2e4eb3b383127a28c711b4ff59
-          ;;
-        arm64)
-          FILE_NAME=go1.16.4.linux-arm64.tar.gz
-          FILE_HASH=8b18eb05ddda2652d69ab1b1dd1f40dd731799f43c6a58b512ad01ae5b5bba21
-          ;;
-        armhf)
-          FILE_NAME=go1.16.4.linux-armv6l.tar.gz
-          FILE_HASH=a53391a800ddec749ee90d38992babb27b95cfb864027350c737b9aa8e069494
-          ;;
-        i386)
-          FILE_NAME=go1.16.4.linux-386.tar.gz
-          FILE_HASH=cd1b146ef6e9006f27dd99e9687773e7fef30e8c985b7d41bff33e955a3bb53a
-          ;;
-      esac
-      # Download the archive, failing on ssl cert problems.
-      curl https://dl.google.com/go/$FILE_NAME -O
-      echo "$FILE_HASH $FILE_NAME" > sha256
-      sha256sum -c sha256 | grep OK
-      tar -C $SNAPCRAFT_STAGE -xf go*.tar.gz --strip-components=1
-    # Don't include any of the files from the go part in the final snap.
-    prime:
-      - "-*"
+    override-pull: |
+      cd $SNAPCRAFT_PROJECT_DIR
+      if [ -f VERSION ]; then
+        PROJECT_VERSION=$(cat VERSION)
+      else
+        PROJECT_VERSION="0.0.0"
+      fi
+      snapcraftctl set-version ${PROJECT_VERSION}
```

Comment:
This whole section can be simplified as:
```

  version:
    plugin: nil
    source: snap/local
    override-pull: |
      cd $SNAPCRAFT_PROJECT_DIR
      GIT_VERSION=$(git describe --tags --abbrev=0 | sed 's/v//')
      snapcraftctl set-version ${GIT_VERSION}
```

The VERSION file method is really only used for versioning docker containers.